UserInterface/SearchBarDemos/SearchBarDemos/SearchBarDemos/SearchBarXamlPage.xaml.cs

```diff
@@ -0,0 +1,24 @@
+﻿using SearchBarDemos.Services;
+using System;
+using System.Windows.Input;
+using Xamarin.Forms;
+
+namespace SearchBarDemos
+{
+    public partial class SearchBarXamlPage : ContentPage
+    {
+        public ICommand PerformSearchCommand { get; set; }
+
+        public SearchBarXamlPage()
+        {
+            InitializeComponent();
+            searchResults.ItemsSource = DataService.Fruits;
+        }
+
+        void OnSearchButtonPressed(object sender, EventArgs e)
+        {
+            SearchBar bar = (SearchBar)sender;
```

Comment:
IMO for the same reason you don't pass the SearchBox to the viewmodel in MVVM: it forms a hard coupling between the otherwise-reusable handler method and a single SearchBar instance. This way the  handler doesn't need to be refactored or altered if other UI/code details change.

Reply:
It seems unlikely that there'd ever be more than one SearchBar on a page?

Reply:
I have some use cases for more than one search bar on the same page, hopefully it will be supported.

Reply:
@davidbritch - Changed
@Italjaard - Multiple searchbars on a single page work fine!